clash-prelude/src/Clash/XException.hs

```diff
@@ -6,14 +6,58 @@ Copyright  :  (C) 2016,      University of Twente,
 License    :  BSD2 (see the file LICENSE)
 Maintainer :  QBayLogic B.V. <devops@qbaylogic.com>
 
-'XException': An exception for uninitialized values
+This module houses utilities to deal with /undefined/ values in Clash. An undefined
+value is translated to @x@ in VHDL and Verilog. At its core, this is handled by
```

Comment:
Is this true? I vaguely remember some clash undefined values becoming "don't care" in VHDL

Reply:
Ooh yes, you're right! It does translate to `-`.

Reply:
In other places it does become X though I think. This is maybe okay, we should probably note that a Clash X is not necessarily the same as X in VHDL or Verilog (since I think outside of BiSignal, we would just treat a Verilog Z as also being X)

This might be fiddly to get right, so I guess keep at it until @DigitalBrains1 doesn't object (since I'm not so rigourous when it comes to doing things properly)